errortypes/severitylevel.go

```diff
@@ -0,0 +1,45 @@
+package errortypes
```

Comment:
A new way to determine error vs warning, so warnings can have their own codes without using a long hardcoded list of which codes are warning vs error (like we had before to a degree).